cmd/ci-reporter/cmd/testgrid.go

```diff
@@ -53,8 +54,13 @@ func (r TestgridReporter) GetCIReporterHead() CIReporterInfo {
 // CollectReportData implementation from CIReporter
 func (r TestgridReporter) CollectReportData(cfg *Config) ([]*CIReportRecord, error) {
 	testgridReportData, err := GetTestgridReportData(*cfg)
+	// if the report data could not get retrieved, log but do not break
 	if err != nil {
-		return nil, err
+		if len(testgridReportData) > 0 {
+			logrus.Infof("Could not retrieve data from some dashboards likely the %s branches have not yet been created", cfg.ReleaseVersion)
+		} else {
+			return nil, err
+		}
 	}
```

Comment:
It's a bit odd from a logical perspective. I think we should either return no error `if len(testgridReportData) > 0` or pre-define an error and check with `errors.Is(…)`.

Reply:
Yes, I agree. It looks like a piece of magic code.

Reply:
I like the `errors,Is(...)` option, but we currently return a multierror which gets put together dynamically, so checking with `errors,Is(...)` will not work now. I will explore options

Reply:
Negate `errors.As` could work in this case I think

Reply:
Can we wrap the multierror conditionally with `fmt.Errorf("%v: %w", err, ErrDashboardNotAvailable)`, like:

```go
package main

import (
	"errors"
	"fmt"
)

func main() {
	err := getErr()
	fmt.Printf("%v\n", errors.Is(err, ErrDashboardNotAvailable))
}

var ErrDashboardNotAvailable = errors.New("bar")

func getErr() error {
	err := errors.New("foo")
	return fmt.Errorf("%v: %w", err, ErrDashboardNotAvailable)
}
```